code/obj/storage/gang_crate.dm

```diff
@@ -167,67 +172,193 @@
+	var/datum/loot_generator/lootMaster
 	level = UNDERFLOOR
 
+
+	proc/initialize_loot_master(x,y)
+		src.vis_controller = new()
+		vis_controller.initialize(src)
+		lootMaster =  new /datum/loot_generator(x,y)
+		RegisterSignal(src, XSIG_MOVABLE_AREA_CHANGED, PROC_REF(alert_gang))
+
 	New()
 		src.AddComponent(/datum/component/log_item_pickup, first_time_only=TRUE, authorized_job=null, message_admins_too=FALSE)
 		..()
 	only_gimmicks
 		New()
 
-			var/datum/loot_generator/lootMaster =  new /datum/loot_generator(3,2)
+			initialize_loot_master(3,2)
 			lootMaster.fill_remaining(src, GIMMICK)
 			..()
 	gear_and_gimmicks
 		New()
-			var/datum/loot_generator/lootMaster =  new /datum/loot_generator(3,2)
+			initialize_loot_master(3,2)
 			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 2)
 			lootMaster.fill_remaining(src, GIMMICK)
 			..()
 
 	guns_and_gear
 		New()
-			var/datum/loot_generator/lootMaster =  new /datum/loot_generator(3,2)
+			initialize_loot_master(3,2)
 			lootMaster.add_random_loot(src, GANG_CRATE_GUN, 1)
 			lootMaster.add_random_loot(src, GANG_CRATE_AMMO, 1)
 			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 2)
 			lootMaster.fill_remaining(src, GIMMICK)
 			..()
 
+	pickup(mob/user)
+		if (open)
+			close()
+		..()
+		if (src.layer == UNDERFLOOR)
+			src.layer = OVERFLOOR
+		var/datum/gang = user.get_gang()
+		if (gang && (trapped || tracking))
+			trapped = FALSE //if one gang member gets their mitts on it, this has done its job
+			toggle_tracking(FALSE)
+			boutput(user, SPAN_ALERT("You disarm the trap in the [src]'s handle. It's now safe to carry."))
+		if (!gang && !open && trapped && owning_gang)
+			if (isliving(user))
+				var/mob/living/H = user
+				idiot = user
+				trapped = FALSE
+				toggle_tracking(FALSE)
+				icon_state = "gang_dufflebag_trap"
+				cant_self_remove = TRUE
+				cant_drop = TRUE
+				var/area/area = get_area(src)
+				playsound(src.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+				boutput(user, SPAN_ALERT("As you pick up \the [src.name], a series of barbs emerge from the handle, lodging in your hand!"))
+				src.owning_gang.broadcast_to_gang("The bag [src.informant] knew about has just been stolen! Looks like it was in \the [area.name]")
+				ON_COOLDOWN(src,"bleed_msg", 30 SECONDS) //set a 30 second timer to remind players to remove this
+				idiot.setStatus("gang_trap", duration = INFINITE_STATUS)
+				H.emote("scream")
+				H.bleeding = max(1,H.bleeding)
+				SPAWN(1 SECOND)
+					bleed_process()
+	handle_other_remove(var/mob/source, var/mob/living/carbon/human/target){
+		var/datum/gang/userGang = source.get_gang()
+		if (cant_drop && idiot && userGang )
+			source.visible_message("[source] rips the [src] right off [target]! Ouch!","You rip the duffle bag from [target]'s hand.")
+			idiot.emote("scream")
+			playsound(idiot.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+			blood_slash(idiot, 2)
+			unhook()
+			return TRUE
+		..()
```

Comment:
```suggestion
		if (!cant_drop || !idiot || !source.get_gang())
			return ..()

		source.visible_message("[source] rips the [src] right off [target]! Ouch!","You rip the duffle bag from [target]'s hand.")
		idiot.emote("scream")
		playsound(idiot.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
		blood_slash(idiot, 2)
		unhook()
		return TRUE
```

Use an early return for this. There is also no need to declare a variable for `source.get_gang()` if it is only being used once, and within an `if` statement in which it may not always be used.